### DISCUSSION TITLE
Source Build Directions for Raspberry Pi 4B?

### DISCUSSION THREAD

**Discussion by pmaine** [AlexandreRouma/SDRPlusPlus#247] [Q&A] [answered]
First of all, I am very excited by this new sdrpp software based upon a few YouTube videos I have watched.
I am very interested in running this software on my Raspberry Pi 4B with 8GB. I have tried to follow the directions for building from source for my architecture and have ran into some issues. I am certain I just need a small amount of assistance to be able to successfully build and run.

I am running the Raspberry Pi OS.
Here are the steps I followed:

Created a directory in my default login directory named "source"
Next cd into the source directory
Next I copied the tarball into the source directory
Next I ran tar xvzf SDRPlusPlus-1.0.3.tar.gz
Next I ran the following commands:
mkdir build
cd build
cmake ..
make -j2

The build appeared to be successful.

Next I try to run the app from the build directory with the following command:
./sdrpp
When the app starts I get the following messages and errors:
[2021-08-13 09:31:23.390] [info] SDR++ v1.0.3
[2021-08-13 09:31:23.392] [info] Loading config
[2021-08-13 09:31:23.394] [warning] ConfigManager locked, waiting...

[2021-08-13 09:31:24.224] [error] Glfw Error 65543: GLX: Failed to create context: GLXBadFBConfig

Thank you in advance for helping me!

**Comment by starvald** [comment]
you will need to edit a file to get it working on a pi4.

core/src/core.cpp (line 275)
change to 
```
const char* glsl_version = "#version 300 es";
glfwWindowHint(GLFW_CLIENT_API, GLFW_OPENGL_ES_API);
glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 1);

```
and rebuild.

**Reply by pmaine** [reply]
Thank you very much!!! The application now runs but when I tune in to a local AM radio station, the audio sputters. I know this issue is off topic for this thread. I'm using an Airspy R2 with a Spyverter. I am attaching a screen snag of my SDRpp running on my Raspberry Pi 4B.
![SDRppOnRaspberryPi4B](https://user-images.githubusercontent.com/5252230/129397281-8ff7da92-f5fd-44a3-948f-08313ad4d8e4.png)

**Comment by starvald** [comment, marked as answer]
yeah have an R2 too and tried it, 10MHz is just too much data for the Pi4 USB, 2.5MHz should be ok though.

**Reply by pmaine** [reply]
That was it! I know that there are many other Raspberry Pi owners that would love to use this software. SDRpp is far better than any of the other open source alternatives. I now need to take a deep dive to learn the features of SDRpp. It looks like it rivals or is better than SDR# and I love that.

Do you think it would be good for me to add a section for the Raspberry Pi in the readme file?

Thanks you soooo much!!!
